src/system/documents/item.ts

```diff
@@ -569,7 +597,7 @@ export class CosmereItem<
      */
     public async use(
         options: CosmereItem.UseOptions = {},
-    ): Promise<D20Roll | [D20Roll, DamageRoll] | null> {
+    ): Promise<D20Roll | [D20Roll, ...DamageRoll[]] | null> {
```

Comment:
It may be more flexible to our future needs for use to just always return an array of `[ Roll ]` (with no specific types of rolls needed to be specified). We may want to have utility rolls that return a formula that isn't a d20 or a damage roll for example. I think it's also better to just have it be a single array of roll objects instead of nested arrays too, because it's easier to filter to get what we want that way.

Reply:
So I did debate making it just a list of Rolls. that would indeed be easier to manage in most areas of the code, and more flexible in future.
Mostly I went this way because I was following Stan's lead, because I also realised that if I swapped to a Roll array, I'd also need to include some degree of identification process to find where various rolls are (to be fair, this should probably also be done in this current set up, but the compile-time checking makes me feel like it is less necessary and we can assume the index locations are right).
I don't know just how many times we'll want the automated roller in the system to be able to handle anything that falls outside of the "Skill roll + potential damage roll" baseline that makes up the majority of Plotweaver's interaction mechanic. It would be good to be ready for it, but is it just belt-and-braces right now?
I'd like a way that "just works" regardless of what is thrown at it, but we'd need more supporting code around that I feel?

As for the nested structure, at this point (and also where it is passed to the chat message) it is becoming un-nested. The "tuple with spread operation" is the only way I could find to define an array where certain types are at certain indexes while still being undefined length. I could have gone with `(D20Roll | DamageRoll)[]` but at that point it might as well be Roll[], this is basically saying the format will be `[D20Roll, DamageRoll, DamageRoll, DamageRoll, ...]`